purchases/src/main/kotlin/com/revenuecat/purchases/deeplinks/DeepLinkHandler.kt

```diff
@@ -0,0 +1,71 @@
+package com.revenuecat.purchases.deeplinks
+
+import android.content.Intent
+import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
+import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.common.debugLog
+import com.revenuecat.purchases.common.verboseLog
+
+/**
+ * Allows to pass deep links to the RevenueCat SDK for processing.
+ */
+@ExperimentalPreviewRevenueCatPurchasesAPI
+class DeepLinkHandler {
+    /**
+     * Represents whether the deep link will be handled by the SDK or not.
+     */
+    enum class HandleResult {
+        /**
+         * Deep link is recognized by RevenueCat's SDK and will be handled.
+         */
+        HANDLED,
+
+        /**
+         * Deep link is recognized by RevenueCat's SDK but it's not configured so it will be cached
+         * and processed after SDK has been configured.
+         */
+        DEFERRED_TO_SDK_CONFIGURATION,
+
+        /**
+         * Deep link is not recognized by RevenueCat's SDK so it's ignored.
+         */
+        IGNORED,
+    }
+
+    companion object {
+        @get:Synchronized
+        internal val cachedLinks = mutableSetOf<DeepLinkParser.DeepLink>()
+
+        /**
+         * Allows the RevenueCat SDK to handle the activity intent to process relevant deep links.
+         * @param intent The intent to process deep links for. Usually obtained from the Activity.
+         * @param shouldCache Whether the deep link should be cached for later processing if the SDK is not configured.
+         */
+        @Suppress("ReturnCount")
+        fun handleDeepLink(intent: Intent, shouldCache: Boolean = true): HandleResult {
```

Comment:
Hmm honestly this was a bit of my preference to try to not bloat the `Purchases` class even more. Specially considering this can actually work without configuring the SDK... But honestly not a strong preference, if you prefer it to be inside `Purchases`, I'm ok moving it. As you said, it keeps everything into a single entry point.